indico/modules/categories/controllers/display.py

```diff
@@ -163,7 +165,7 @@ def _process(self):
         query = (Category.query
                  .filter(Category.title_matches(q))
                  .options(undefer('deep_children_count'), undefer('deep_events_count'), undefer('has_events'),
-                          joinedload('acl_entries')))
+                          undefer('has_children'), joinedload('acl_entries')))
```

Comment:
Not sure If I should add `undefer(has_children)` in other places such as: https://github.com/indico/indico/pull/4980/files#diff-791a87a96ed182d72dbd69f1d61092083942dd060d291db0b1611abe5e84e729R212

Reply:
basically code that results in calling `serialize_category` should undefer it when querying the category passed to it to avoid triggering the extra query

Reply:
Thanks! That makes sense.. In that case I think I've covered everything.